packages/core/injector/helpers/provider-classifier.ts

```diff
@@ -0,0 +1,20 @@
+import {
+  ClassProvider,
+  FactoryProvider,
+  Provider,
+  ValueProvider,
+} from '@nestjs/common';
+import { isUndefined } from '@nestjs/common/utils/shared.utils';
+
+export function isClassProvider(provider: Provider): boolean {
```

Comment:
I guess we could use ts guards here like

```suggestion
export function isClassProvider<T = any>(provider: Provider): provider is ClassProvider<T> {
```

Reply:
Good suggestion Micael! Thanks 👍